.github/workflows/build.yml

```diff
@@ -7,57 +7,68 @@ on:
       # Push events on the main branch
       - main
       - release/**
+      - set-product-version
 
 env:
   PKG_NAME: consul
+  VERSION: oss
```

Comment:
This is `METADATA` in ENT.  IMHO I like that terminology better since it's competing with a lot of other "versions" here.